sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetadataCacheSuite.scala

```diff
@@ -126,4 +129,39 @@ class HiveMetadataCacheSuite extends QueryTest with SQLTestUtils with TestHiveSi
   for (pruningEnabled <- Seq(true, false)) {
     testCaching(pruningEnabled)
   }
+
+  test("cache TTL") {
+    val sparkConfWithTTl = new SparkConf().set(SQLConf.METADATA_CACHE_TTL.key, "1")
+    val newSession = SparkSession.builder.config(sparkConfWithTTl).getOrCreate().cloneSession()
+
+    withSparkSession(newSession) { implicit spark =>
+      withTable("test_ttl") {
+        withTempDir { dir =>
+          spark.sql(s"""
+            |create external table test_ttl (id long)
+            |partitioned by (f1 int, f2 int)
+            |stored as parquet
+            |location "${dir.toURI}"""".stripMargin)
+
+          val tableIdentifier = TableIdentifier("test_ttl", Some("default"))
+
+          // First, make sure the test table is not cached.
+          assert(getCachedDataSourceTable(tableIdentifier) === null)
+          // This query will make the table cached.
+          spark.sql("select * from test_ttl")
+          assert(getCachedDataSourceTable(tableIdentifier) !== null)
+          // Wait until the cache expiration.
+          Thread.sleep(1500L) // 1.5 seconds > 1 second.
+          // And the cache is gone.
+          assert(getCachedDataSourceTable(tableIdentifier) === null)
+        }
+      }
+    }
+  }
+
+  private def getCachedDataSourceTable(table: TableIdentifier)
+                                      (implicit spark: SparkSession): LogicalPlan = {
```

Comment:
nit format:
```
  private def getCachedDataSourceTable(table: TableIdentifier)
      (implicit spark: SparkSession): LogicalPlan = {
```

Reply:
Fixed in https://github.com/apache/spark/pull/28852/commits/7d94f943496fb463f18cd445ccbde3f5225c8e7b